connection_test.go

```diff
@@ -1322,3 +1322,27 @@ func TestInvalidTransportHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomDialer(t *testing.T) {
+	sopts := testutils.NewOpts()
```

Comment:
you can just pass `nil` to `WithTestServer` and it'll use "default" opts which is what you're passing right now